acvm_backend_barretenberg/src/pedersen.rs

```diff
@@ -1,12 +1,59 @@
 use common::acvm::FieldElement;
-use wasmer::Value;
 
-use common::barretenberg_structures::Assignments;
+use super::Barretenberg;
 
-use super::{Barretenberg, FIELD_BYTES};
+pub(crate) trait Pedersen {
+    fn compress_native(&self, left: &FieldElement, right: &FieldElement) -> FieldElement;
+    fn compress_many(&self, inputs: Vec<FieldElement>) -> FieldElement;
+    fn encrypt(&self, inputs: Vec<FieldElement>) -> (FieldElement, FieldElement);
+}
+
+#[cfg(feature = "native")]
+impl Pedersen for Barretenberg {
+    fn compress_native(&self, left: &FieldElement, right: &FieldElement) -> FieldElement {
+        let result_bytes = barretenberg_sys::pedersen::compress_native(
+            left.to_be_bytes().as_slice().try_into().unwrap(),
+            right.to_be_bytes().as_slice().try_into().unwrap(),
+        );
+
+        FieldElement::from_be_bytes_reduce(&result_bytes)
+    }
+
+    #[allow(dead_code)]
```

Comment:
?

Reply:
The code was already dead, we're just not making it public anymore. (causing the lint)

Reply:
Where are you seeing this? I removed all the dead code annotations and I'm not getting any clippy alerts.

Reply:
![image](https://user-images.githubusercontent.com/15848336/234377033-2652d6d1-3508-4944-9ef0-b0468674a9fb.png)

Reply:
It'll only appear on the version corresponding to the feature flag currently being compiled. Maybe you removed it from an inactive block?

Reply:
Yeah, I'm not sure. I suppose we can leave them